### PR TITLE
Fix CI/CD to ensure tests run

### PR DESCRIPTION
**Pull request by eveleighoj**
There have been multiple failing deployments recently to all of the environments for this application. When examine the repo you can see that acceptance tests have been consistently failing on the main branch as well as development branches.

Under more examination tests were being ran after deployments have started so even if tests fail the app was still be deployed which isn't right. This PR aims to replicate the behaviour in digital-land.info which stops this from happening.

Note(@rosado ): this PR now includes changes from #158 and the changes above have largely rewritten and include: 

- Cleanup in github workflows: there's one `base.yaml` workflow that calls `test.yaml` and `deploy.yml` () workflows. Extracted common steps to custom actions in `.github/actions`. 
- the 'deploy.yml' is triggered from `base.yaml` only on 'main' branch
- updated used versions of Node and github actions
- included unit and integration test steps in the 'test' workflow (it was running only the acceptance tests till now)

Note(@eveleighoj):
- this also adds a fix for this [issue](https://trello.com/c/YzGXA0Dd/3428-bug-check-service-failure-due-to-404-response-from-api-in-all-environments)

### CODE REVIEW

File: .github/workflows/deploy-to-dev.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+          sudo ./aws/install --update
+          sudo apt-get update
+          sudo apt-get install -y rsync
+
+      - id: install-node-dependencies
+        run: |
+          npm ci
+          npx playwright install --with-deps
+
+      - uses: aws-actions/configure-aws-credentials@v1-node16
```
Comment (rosado) [inline .github/workflows/deploy-to-dev.yml:35]: any reason we're using an ancient version of this action?

File: .github/workflows/deploy-to-dev.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        with:
+          aws-access-key-id: ${{ secrets.DEPLOY_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.DEPLOY_AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      - uses: aws-actions/amazon-ecr-login@v1
```
Comment (rosado) [inline .github/workflows/deploy-to-dev.yml:41]: same as above, why not latest version?

File: config/index.js [hunk 3]
```diff
@@ -19,7 +20,7 @@ const getConfig = () => {
 
   const customConfig = readConfig(environment)
 
-  const combinedConfig = Object.assign({}, defaultConfig, customConfig)
+  const combinedConfig = _.merge({}, defaultConfig, customConfig)
```
Comment (cpcundill) [inline config/index.js:23]: Key line to fix the 404 issue seen in production!

**Review by cpcundill — APPROVED** [review]

File: config/index.js (hunk 3 above)
Comment (rosado) [reply config/index.js:23]: PR with config schema & validation coming soon. This will guard against using obviously invalid configuration.

**Pull request merged**